lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -250,34 +250,36 @@ bool SessionStateEnforcer::handle_session_init_rule_updates(
  * start terminating session process
  */
 bool SessionStateEnforcer::m5g_release_session(
-    SessionMap& session_map, const std::string& imsi, const std::string& dnn,
+    SessionMap& session_map, const std::string& imsi, const uint32_t& pdu_id,
     SessionUpdate& session_update) {
   /* Search with session search criteria of IMSI and apn/dnn and
    * find  respective sesion to release operation
    */
-  SessionSearchCriteria criteria(imsi, IMSI_AND_APN, dnn);
+  SessionSearchCriteria criteria(imsi, IMSI_AND_PDUID, pdu_id);
   auto session_it = session_store_.find_session(session_map, criteria);
   if (!session_it) {
     MLOG(MERROR) << "No session found in SessionMap for IMSI " << imsi
-                 << " with DNN " << dnn << " to release";
+                 << " with PDU " << pdu_id << " to release";
     return false;
   }
   // Found the respective session to be updated
   auto& session   = **session_it;
   auto session_id = session->get_session_id();
   /*Irrespective of any State of Session, release and terminate*/
   SessionStateUpdateCriteria& session_uc = session_update[imsi][session_id];
-  MLOG(MINFO) << "Trying to release session with id " << session_id
-              << " from state "
-              << session_fsm_state_to_str(session->get_state());
-  m5g_start_session_termination(imsi, session, dnn, &session_uc);
+  MLOG(MDEBUG) << "Trying to release session with id " << session_id
+               << " from state "
+               << session_fsm_state_to_str(session->get_state());
+  m5g_start_session_termination(
+      session_map, imsi, session, pdu_id, &session_uc);
   return true;
 }
 
 /*Start processing to terminate respective session requested from AMF*/
 void SessionStateEnforcer::m5g_start_session_termination(
-    const std::string& imsi, const std::unique_ptr<SessionState>& session,
-    const std::string& dnn, SessionStateUpdateCriteria* session_uc) {
+    SessionMap& session_map, const std::string& imsi,

```

Comment:
Let's remove `const std::string& imsi,` from the params since we can easily get it by doing `session->get_imsi()`

Reply:
Yes, Changes done.